File: Data Structures & Algorithms/binary-search/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let s = 0;
    let e = nums.length - 1;
    let mid = Math.floor((s + e )/2);

    while (s <= e) {

        if (nums[mid] === target) {
            return mid;
        } else if (nums[mid] > target) {
            e = mid - 1;
        } else {
            s = mid + 1;
        }

        mid = Math.floor((s + e )/2);
    }

    return -1;
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-0.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        n = len(nums)
        size = n * 2
        ans = size * [None]

        for i in range(len(nums)):
            ans[i] = nums[i]
            ans[i + n] = nums[i]
        
        return ans